pytest_asyncio/plugin.py

```diff
@@ -249,14 +248,38 @@ def _add_kwargs(
     return ret
 
 
-def _wrap_asyncgen(func: Callable[..., AsyncIterator[_R]]) -> Callable[..., _R]:
-    @functools.wraps(func)
+def _perhaps_rebind_fixture_func(
```

Comment:
You described two ways to fix the issue in #197: Adjusting `_wrap_async` so that it looks like a bound method to pytest or reimplement pytest's method rebinding.

Can you share the reasoning why you went with reimplementing the method rebinding rather than adding `__self__` to `_wrap_async_`?

Reply:
This was the more elegant path, I felt. Not a very satisfying answer, as it is subjective. Mostly, I wasn't sure I understood all the edgecases of the interaction between pytest, the wrapper, and both bound and unbound fixture functions. By using this approach you can follow how the fixture function is rebound or not entirely within the pytest-asyncio codebase, so the process is easier to follow and reason about.

Reply:
That is a perfectly satisfying answer to me, thank you.